idb/idb.go

```diff
@@ -381,6 +381,11 @@ type IndexerDbOptions struct {
 	// This means the total number of active queries that can be running
 	// concurrently can never be more than this
 	MaxConn uint32
+
+	IndexerDatadir string
+	AlgodDataDir   string
+	AlgodToken     string
+	AlgodAddr      string
```

Comment:
Technically `AlgodDataDir` resolves to `AlgodToken` / `AlgodAddr`, so we could remove `AlgodDataDir`.

Reply:
I thought both were needed since there are 2 ways for the fetcher to create an algod client, https://github.com/algorand/indexer/blob/df6bcb6ced21a354bbc80daf41642c81226c1d40/idb/postgres/internal/migrations/local_ledger/migration.go#L50-L63

Reply:
The `ForDataDir` would do something like resolve the address/token and then call `ForNetAndToken`